service-loadbalancer/service_loadbalancer.go

```diff
@@ -348,14 +360,35 @@ func (lbc *loadBalancerController) getEndpoints(
 	return
 }
 
-// encapsulates all the hacky convenience type name modifications for lb rules.
-// - :80 services don't need a :80 postfix
-// - default ns should be accessible without /ns/name (when we have /ns support)
-func getServiceNameForLBRule(s *api.Service, servicePort int) string {
-	if servicePort == 80 {
-		return s.Name
+// LBRule encapsulates a rule object compatible with Ingress to allow custom routing rules
+type LBRule struct {
+	ACL  string
+	URLs []string
+}
+
+type RuleGetter interface {
+	getRules(s *api.Service, servicePort int) *LBRule
```

Comment:
Either make this a private interface or make the getRules method public. 
Also suggest:
// RuleGetter translates a service:port combination to a loadbalancer rule. Eg: foo, 8080 becomes acl rule ...